sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/jdbc/JDBCTable.scala

```diff
@@ -0,0 +1,129 @@
+
+  override def name(): String = ident.toString
+
+  override def capabilities(): util.Set[TableCapability] = {
+    import TableCapability._
+    Set(BATCH_READ, V1_BATCH_WRITE, TRUNCATE).asJava
+  }
+
+  override def newScanBuilder(options: CaseInsensitiveStringMap): ScanBuilder = {
+    val mergedOptions = new JDBCOptions(
+      jdbcOptions.parameters.originalMap ++ options.asCaseSensitiveMap().asScala)
+    val session = SparkSession.active
+    val resolver = session.sessionState.conf.resolver
+    val timeZoneId = session.sessionState.conf.sessionLocalTimeZone
+    val parts = JDBCRelation.columnPartition(schema, resolver, timeZoneId, mergedOptions)
+    new JDBCScan(JDBCRelation(schema, parts, mergedOptions)(session))
+  }
+
+  override def newWriteBuilder(info: LogicalWriteInfo): WriteBuilder = {
+    val mergedOptions = new JdbcOptionsInWrite(
+      jdbcOptions.parameters.originalMap ++ info.options.asCaseSensitiveMap().asScala)
+    new JDBCWrite(schema, mergedOptions)
+  }
+}
+
+class JDBCScan(relation: JDBCRelation) extends ScanBuilder with V1Scan
+  with SupportsPushDownFilters with SupportsPushDownRequiredColumns {
+
+  private var pushed = Array.empty[Filter]
+
+  override def pushFilters(filters: Array[Filter]): Array[Filter] = {
+    val options = relation.jdbcOptions
+    if (options.pushDownPredicate) {
+      val dialect = JdbcDialects.get(options.url)
+      val (pushed, unSupported) = filters.partition(JDBCRDD.compileFilter(_, dialect).isDefined)
+      this.pushed = pushed
+      unSupported
+    } else {
+      filters
+    }
+  }
+
+  override def pushedFilters(): Array[Filter] = pushed
+
+  private var prunedSchema = relation.schema
+
+  override def pruneColumns(requiredSchema: StructType): Unit = {
+    // JDBC doesn't support nested column pruning.
+    val requiredCols = requiredSchema.map(_.name)
+    prunedSchema = StructType(relation.schema.fields.filter(f => requiredCols.contains(f.name)))
+  }
+
+  override def readSchema(): StructType = prunedSchema
+
+  override def build(): Scan = this
+
+  override def toV1TableScan[T <: BaseRelation with TableScan](context: SQLContext): T = {
+    new BaseRelation with TableScan {
+      override def sqlContext: SQLContext = context
+      override def schema: StructType = prunedSchema
+      override def needConversion: Boolean = relation.needConversion
+      override def buildScan(): RDD[Row] = {
+        relation.buildScan(prunedSchema.map(_.name).toArray, pushed)
+      }
+    }.asInstanceOf[T]
+  }
+}
+
+class JDBCWrite(schema: StructType, options: JdbcOptionsInWrite) extends V1WriteBuilder
+  with SupportsTruncate {
+
+  private var isTruncate = false
+
+  override def truncate(): WriteBuilder = {
+    isTruncate = true
+    this
+  }
+
+  override def buildForV1Write(): InsertableRelation = new InsertableRelation {
+    override def insert(data: DataFrame, overwrite: Boolean): Unit = {
+      // TODO: do truncate and append atomically.
```

Comment:
If you don't mind, please file a JIRA issue and make this IDed TODO like `TODO(SPARK-XXX)`.